crates/contracts/src/deploy.rs

```diff
@@ -0,0 +1,164 @@
+use crate::{
+    BalancerV2Authorizer, BalancerV2Vault, GPv2AllowListAuthentication, GPv2Settlement,
+    UniswapV2Factory, UniswapV2Router02, WETH9,
+};
+use anyhow::{Context, Result};
+use ethcontract::{dyns::DynTransport, Address, Web3, U256};
+use std::path::{Path, PathBuf};
+
+pub struct Contracts {
+    pub balancer_vault: BalancerV2Vault,
```

Comment:
Probalby needs to be 
```suggestion
    pub balancer_vault: Option<BalancerV2Vault>,
```

Since its not available on xDai.